### ISSUE TITLE
New ZIM request: ReadTheDocs dot org

### ISSUE THREAD

**Issue by n8willis** [openzim/zim-requests#446]
- Website URL: https://readthedocs.org/
- License: CC? I think it's possible there are multiple options for individual projects that use the service
- Desired ZIM Title: Read The Docs
- Desired ZIM Description: Public documentation from a large assortment of FOSS software projects
- Language (ISO 639-3): **eng**
- Desired Main Page (homepage, if different from website URL): **n/a**
- Is this a MediaWiki?: **no**

ReadTheDocs hosts the official documentation for a large assortment of open-source software projects, from applications to frameworks/libraries down to programming languages themselves. As such, it's an invaluable resource when developing, studying, or troubleshooting, and is sorely missed when one is offline.

I'm not sure I have a complete picture of all the size/complexity issues, but I pinged the project team about that on Twitter, and there are some year-in-review blog posts: https://blog.readthedocs.com/archive/tag/year-in-review/ that put the total number of *projects* using the service at ~250,000 or so. I don't know how to guesstimate the text/image ratio.

Notably, the hosting site also supports versioning (in the per-project namespaces), so the docs for a long-lived Python library might actually add up to 50 versions. But I do think that's important to the value of the project; if you're debugging code from a prior release it wouldn't do a lot of good to only have access to later versions of the documentation.

I also feel like they may support more than one license option, in order to accommodate the preferred/in-use licenses of each project, but they do require that the project be open-source, which ought to provide some level of baseline.

Finally, there is a "private repo" service that the project charges money for and which (naturally) does not require open licensing or public access at all. This request doesn't cover that.

**Comment by B100GH** [comment]
This would be great since docs for code is super useful online.

**Comment by Popolechien** [comment]
@n8willis Have they replied to your Twitter ping?

I don't think we can cover versioning without building a dedicated scraper, but let's see first if we can clarify the licensing.

**Comment by n8willis** [comment]
> @n8willis Have they replied to your Twitter ping?

That's where I got the blog link.


> I don't think we can cover versioning without building a dedicated scraper, but let's see first if we can clarify the licensing.

Long-term, me personally pinging the Twitter account isn't the path forward. They're not a large team; they are on GitHub as well (https://github.com/readthedocs), so engaging with them directly to get technical questions answered ought to be done that way.

As far as the versioning referred to in the original report is concerned, lemme be clearer; that's not a matter of preserving edit history: docs repos have versions built into the URI hierarchy: read-the-docs-site-/SomeLibrary/1.0/index.html , read-the-docs-site-/SomeLibrary/1.2/index.html, read-the-docs-site-/SomeLibrary/2.0/index.html; that kind of thing.

Including that is a user-level benefit; I mentioned it to provide better context for the "how large is the library" question.

**Comment by rgaudin** [comment]
Is there a list or API to get the list of hosted projects?

**Comment by n8willis** [comment]
> Is there a list or API to get the list of hosted projects?

@rgaudin I'm not clear if you were directing this question at me, personally, but, if so, I must reiterate that the source of technical answers on @readthedocs's details genuinely will need to be the ReadTheDocs team on Github.

**Comment by rgaudin** [comment]
@n8willis yeah we'll have to do that. It's just that as a regular user, I'm always going to rtd via projects' repos so I am unaware of a central list of all projects (80K says the homepage) ; which would be required to build a scraper.

[their doc](https://dev.readthedocs.io) doesn't indicate any API so this will have to go through them in some way.

**Comment by n8willis** [comment]
I know that this new-zim-request is big in that it isn't an "another generic mediawiki site" one, but it's big because the requested site is important and serves its particular audience in the best way that it can.

I really believe that if you engage directly with the engineering (and perhaps the decison-makin') folks at ReadTheDocs the two of you would quite quickly be able to hash out whether / where there are impedance mismatches between your respective codebases. And, given that both teams (in my limited experience with them) are motivated by quite similar goals, there's a fairly high likelihood that the conversation would prove useful, even if you do encounter some technical hurdle at the initial contact.

I'm only pointing out that it would be a colossal win for end users if you were to do it, and hope that, between the two crews, you're able to.

**Comment by Popolechien** [comment]
Pinging @ericholscher then.

**Comment by ericholscher** [comment]
Not sure what this is about? Is there a link with more context about zim and what this issue is trying to achieve?

**Comment by rgaudin** [comment]
@ericholscher, [https://wiki.openzim.org/wiki/OpenZIM](ZIM is a file format) for offline distribution of content, through the use of a Reader App ([https://kiwix.org](Kiwix) mostly).

We provide ZIM files for many free-to-distribute content such as Wikipedia, TED Talks, Stack Exchanges, Gutenberg, etc. See [List of Content](https://wiki.kiwix.org/wiki/Content_in_all_languages) or [ZIM files to download](http://download.kiwix.org/zim/) or, probably even better, our [Online Demo Library](http://library.kiwix.org/?lang=eng&q=ready).

@n8willis suggested that we create a ZIM file of the whole Read The Docs. I think there's an agreement that this would be a great addition but we'd need some clarifications on Licenses (I believe those are per-project), projects lists/discovery and should the former be cleared-out, the authorization to scrape ReadTheDocs (if this is the appropriate technical way forward – we don't always scrape websites remotely).

**Comment by RavanJAltaie** [comment]
@rgaudin @Popolechien 
I'm quoting from their website:
" Copyright infringement and DMCA policy[](https://docs.readthedocs.io/en/stable/terms-of-service.html#copyright-infringement-and-dmca-policy)
If you believe that content on our website violates your copyright or other rights, please contact us in accordance with our [Digital Millennium Copyright Act Policy](https://docs.readthedocs.io/en/stable/dmca/index.html). There may be legal consequences for sending a false or frivolous takedown notice. Before sending a takedown request, you must consider legal uses such as fair use and licensed uses.

We will terminate the Accounts of repeat infringers of this policy.

Intellectual property notice[](https://docs.readthedocs.io/en/stable/terms-of-service.html#intellectual-property-notice)
Short version: We own the Service and all of our Content. In order for you to use our Content, we give you certain rights to it, but you may only use our Content in the way we have allowed."

I believe it's a complicated process to distinguish the allowed project and therefore scrape them independently.
I'm closing the issue, feel free to re open the issue if you have another opinion.
